lib/hawkular/base_client.rb

```diff
@@ -163,6 +163,9 @@ def initialize(message, status_code = 0)
       attr_reader :message, :status_code
     end
 
+    class HawkularConnectionException < HawkularException
```

Comment:
What is the benefit in adding a custom exception for the gem? it indicates a standard connection error that is not unique for hawkular (or to be more accurate, multiple different situations of connection issues)